src/lib/ui/thermostat/ClimatePreset.tsx

```diff
@@ -0,0 +1,299 @@
+    } else {
+      setValue('heatPoint', undefined)
+      setValue('coolPoint', undefined)
+    }
+  }
+
+  const allPresets = useMemo(
+    () => device.properties.available_climate_presets ?? [],
+    [device]
+  )
+
+  const otherPresets = useMemo(
+    () =>
+      allPresets.filter(
+        (other) => other.climate_preset_key !== preset.climate_preset_key
+      ),
+    [allPresets, preset]
+  )
+
+  const isCreate = _preset == null
+  const createMutation = useCreateThermostatClimatePreset()
+  const updateMutation = useUpdateThermostatClimatePreset()
+  const loading = isCreate ? createMutation.isPending : updateMutation.isPending
+
+  const onSubmit = (): void => {
+    if (
+      isCreate &&
+      otherPresets.some((other) => other.climate_preset_key === state.key)
+    ) {
+      setError('key', {
+        type: 'validate',
+        message: 'Preset with this key already exists.',
+      })
+      return
+    }
+
+    const name = state.name.replace(/\s+/g, ' ').trim()
+
+    const body = {
+      climate_preset_key: state.key,
+      device_id: device.device_id,
+      name: name === '' ? undefined : name,
+      cooling_set_point_fahrenheit: state.coolPoint,
+      heating_set_point_fahrenheit: state.heatPoint,
+      fan_mode_setting: state.fanMode,
+      cooling_set_point_celsius:
+        typeof state.coolPoint === 'number'
+          ? toCelcius(state.coolPoint)
+          : undefined,
+      heating_set_point_celsius:
+        typeof state.heatPoint === 'number'
+          ? toCelcius(state.heatPoint)
+          : undefined,
+      hvac_mode_setting: state.hvacMode,
+    }
+
+    if (isCreate) {
+      createMutation.mutate(body)
+    } else {
+      updateMutation.mutate({
+        ...body,
+        manual_override_allowed: true,
+      })
+    }
+  }
+
+  return (
+    <div
+      {...attrs}
+      className={classNames('seam-thermostat-climate-preset', attrs.className)}
+    >
+      <ContentHeader title={preset.display_name} onBack={onBack} />
+
+      <div className='seam-main'>
+        <form
+          onSubmit={(e) => {
+            void handleSubmit(onSubmit)(e)
+          }}
+        >
+          {isCreate && (
+            <FormField>
+              <InputLabel>Key</InputLabel>
+              <TextField
+                size='large'
+                clearable
+                hasError={errors.key != null}
+                helperText={errors.key?.message}
+                inputProps={{
+                  ...register('key', {
+                    required: 'required',
+                    maxLength: {
+                      value: 20,
+                      message: 'max 20 chars',
+                    },
+                    minLength: {
+                      value: 3,
+                      message: 'min 3 chars',
+                    },
+                    validate(value) {
+                      const fixedValue = value.replace(/\s+/g, '').trim()
```

Comment:
What is this for? Fixed? Or normalized?

Reply:
yes, normalized. I forgot the correct naming here